lib/vhdl-linter.ts

```diff
@@ -161,40 +163,35 @@ export class VhdlLinter {
     }
     let start;
     let i = 0;
-    if (this.file) {
-      start = Date.now();
-      try {
-        await Elaborate.elaborate(this);
-        if (profiling) {
-          console.log(`check ${i++}: ${Date.now() - start}ms`);
-          start = Date.now();
-        }
-        // await this.removeBrokenActuals();
+    start = Date.now();
+    try {
+      await Elaborate.elaborate(this);
+      if (profiling) {
+        console.log(`check ${i++}: ${Date.now() - start}ms`);
+        start = Date.now();
+      }
+      // await this.removeBrokenActuals();
+      if (profiling) {
+        console.log(`check ${i++}: ${Date.now() - start}ms`);
+        start = Date.now();
+      }
+      for (const checkerClass of rules) {
+        const checker = new checkerClass(this);
+        checker.check();
         if (profiling) {
```

Comment:
this profiling might be misleading when an async operation is running...

Reply:
I dont understand your point?

Reply:
Some of the check functions are async and the promise is not awaited on.

Reply:
Which ones? that should not be possible anymore?

Reply:
The following:
`lib/rules/library.ts`
`lib/rules/not-declared.ts`
`lib/rules/port-declaration.ts`
`lib/rules/type-resolved.ts`
`lib/rules/unused.ts`